the-collab-lab/tcl-43-smart-shopping-list
# Added Firebase as a dependency + Added send data functionality + rece…

## Description

This PR is updating the dependencies to add in Firebase, as well as adding in two new components `AddItem` and `List`. The `List` component uses the `onSnapshot` method from Firestore to listen to the document so that each time the content changes, it will update the data in the app. The `AddItem` component built the basic initial functionality to hook up the project so that any changes made in the app will update the data in Firestore. We did this by using the `addDoc` method from Firestore.

We decided to add these functionalities into two new components as we determined that they would be different views for the user within the app. 

## Related Issue

Closes #1 

## Acceptance Criteria

- [X] [firebase](https://www.npmjs.com/package/firebase) has been added as a project dependency
- [X] Making a change in the Firestore database will update the data in the app
- [X] Making a change in the app will update the data in Firestore

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

Started from scratch

### After

The UI is very simple as we were focusing on the functionality and not styling.
<img width="278" alt="Screen Shot 2022-04-08 at 11 27 19 AM" src="https://user-images.githubusercontent.com/54180641/162491931-ba0bf854-8d9b-4ae3-b59b-395aafb40fa8.png">

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with `git pull origin sk-rt-connect-to-firestore` and check that branch out with `git checkout sk-rt-connect-to-firestore`
- Use `npm install` to install the newly added dependencies locally and `npm start` to launch the app.
- Click on the "Click to submit new item" button and make sure it adds an item to the list (should be another "banana"). Open up the database to check that it is appearing in the list of items collection [here](https://console.firebase.google.com/u/0/project/tcl-43-smart-shopping-list/firestore/data/~2FShopping-List~2FwxGXR3Kk93w0tMObuqAl). 
- While in the Firestore Database, click on "Add document" under the Shopping-List column. For the document, ID click on the blue "Auto-ID". For field, enter in `item` and add a value of an item you'd like to add to the shopping list! Click save, then navigate back to your browser tab that has the app running, and you should see your list item appear!
